### PR TITLE
FIX Proxy errors and tests

### PR DESCRIPTION
**Pull request by dmoranj**
Fix an error with the proxy functionality that produced an error along with the redirected response (due to a syntax error in one function of the code).

It was uncovered thanks to a bug discovered in LightweightM2M Library by @ezequiel-umu, in the following issue:

https://github.com/telefonicaid/lwm2m-node-lib/issues/93

### CODE REVIEW

File: lib/middlewares.js [hunk 1]
```diff
@@ -51,7 +51,11 @@ function proxyRequest(request, next) {
 }
 
 function isObserve(packet) {
-  request.packet.options.reduce(or, false);
+  function isObserve(option) {
+    return option.name === 'Observe';
+  }
```
Comment (mcollina) [inline lib/middlewares.js]: Can you move this outside the `isObserve` function?

**Comment by dmoranj** [comment]
Tests were fixed refactoring the afterEach() a little bit (making it asynchronous and waiting for all the close events to happen). 

It has been tested in the following versions successfully:
- v0.10.42
- v0.12.10
- v4.3.1

Most of the tests fail for v5, as I explained in a previous post in #103

File: lib/middlewares.js (hunk 1 above)
Comment (dmoranj) [inline lib/middlewares.js]: Done in 155e2e7

**Comment by dmoranj** [comment]
Concerning Travis checks, as you can see, jus one fails, the one corresponding to v5.

**Comment by dmoranj** [comment]
@mcollina Any trouble with the PR, or should I merge it?

File: package.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
     "pre-commit": "0.0.9",
     "chai": "~3.4.1",
     "mocha": "~2.3.4",
-    "timekeeper": "0.0.5",
+    "timekeeper": "*",
```
Comment (mcollina) [inline package.json]: Please depend with the ^ or ~

**Comment by mcollina** [comment]
ok with a nit. go ahead, merge and release

**Comment by dmoranj** [comment]
Fixed. I'll merge the PR, but I'm not entirely sure I can make the release myself.

**Comment by mcollina** [comment]
Can you put node v5 in? it should work now.

I'll release this as 0.14.0, or do think this is a patch release?

**Pull request merged**